### ISSUE TITLE
radar 2.0 crashes

### ISSUE THREAD

**Issue by axelrathey** [iobroker-community-adapters/ioBroker.radar2#35]
radar 2.0 does not work any more. It becomes shortly green, then it went to red.

There are some debug messages von radar 2.0

Thanks for your help

adar2.0	2020-02-23 14:19:27.010	info	(17800) radar2 set to scan every 20 seconds and printers every 720 minutes.
radar2.0	2020-02-23 14:19:27.008	info	(17800) arp-scan will use the following interfaces: [ 'br0' ]
radar2.0	2020-02-23 14:19:27.005	info	(17800) Remove name end for host names: .fritz.box
radar2.0	2020-02-23 14:19:27.003	info	(17800) use known IP list: [ '1.1.1.1' ]
radar2.0	2020-02-23 14:19:27.000	info	(17800) use known BT list: [ '01:12:23:34:45:56' ]
radar2.0	2020-02-23 14:19:26.995	info	(17800) radar2 set to flag items away if they are not seen for 2 minutes
radar2.0	2020-02-23 14:19:26.993	info	(17800) found 'node-bluetooth'
radar2.0	2020-02-23 14:19:26.961	info	(17800) found '@abandonware/noble'
radar2.0	2020-02-23 14:19:26.425	info	(17800) Adapter disconnected and stopped with dostop(false) and callback(true)
radar2.0	2020-02-23 14:19:26.419	error	at process._tickCallback (internal/process/next_tick.js:63:19)
radar2.0	2020-02-23 14:19:26.419	error	at state.handle.lookup (dgram.js:242:18)
radar2.0	2020-02-23 14:19:26.419	error	(17800) Error: bind EACCES 0.0.0.0:67
radar2.0	2020-02-23 14:19:26.417	error	(17800) uncaught exception: bind EACCES 0.0.0.0:67
radar2.0	2020-02-23 14:19:25.829	info	(17800) net-ping not available! Will try to use normal ping!
radar2.0	2020-02-23 14:19:25.797	info	(17800) radar2 initialization started...
radar2.0	2020-02-23 14:19:25.583	info	(17800) starting. Version 1.0.9 in /opt/iobroker/node_modules/iobroker.radar2, node: v10.19.0

**Comment by martingarber** [comment]
same problem here after update node.js to 12.16.3 and npm to 6.14.4

**Comment by frankjoke** [comment]
As listed in readme, you need to rerun the scripts after updating node or even after updating system with some new versions of BT or other drivers because the new versions do not have authority!

**Comment by axelrathey** [comment]
what do you mean with rerun scripts?

**Comment by frankjoke** [comment]
In the readme of the adapter you find installation section where you are asked to do this:
`sudo apt-get install -y coreutils libcap2-bin arp-scan bluetooth bluez libbluetooth-dev libudev-dev net-tools
# and below need to be run whenever you update nodejs!
sudo setcap cap_net_admin,cap_net_raw,cap_net_bind_service=+eip $(eval readlink -f `which arp-scan`)
sudo setcap cap_net_admin,cap_net_raw,cap_net_bind_service=+eip $(eval readlink -f `which node`)
sudo setcap cap_net_admin,cap_net_raw,cap_net_bind_service=+eip $(eval readlink -f `which arp`)
sudo setcap cap_net_admin,cap_net_raw,cap_net_bind_service=+eip $(eval readlink -f `which hcitool`)
sudo setcap cap_net_admin,cap_net_raw,cap_net_bind_service=+eip $(eval readlink -f `which hciconfig`)
sudo setcap cap_net_admin,cap_net_raw,cap_net_bind_service=+eip $(eval readlink -f `which l2ping`)`

You need to rerun the commands below the second line!

**Comment by martingarber** [comment]
Hi Frank,

ok. I'll try this. Thanks!!
Have you any news about the adapter without bluetooth device? I've a virtual machine without bluetooth device. Unfortunately the adapter crashes because of this.

thank you in advance

best regards

**Comment by frankjoke** [comment]
I tested this on one of my VMs without BT and it works if you switch noble off (use hcitool instead of noble) for me. Did you test this?

**Comment by martingarber** [comment]
Hi,

it works both (without bluetooth and after nodejs update). Thank you very much!!

best regards

**Comment by axelrathey** [comment]
Hi,
It works for me now too. 

First I copied the above setcap lines but there were some missing " ' ".
So I copied them from the radar2 readme in github. After this restart radar2 failed again.

Then I switched from noble to hcitool in the radar2 config and now radar2 works fine.

My BananaPI M3 has BT onboard, but this BT is not supported with raspbian on the bananapi. 

Thanks for your help
Axel
